### PR TITLE
fix(create): add default sw in create template

### PR DESCRIPTION
**Pull request by anshumanv**
add default sw in create template

ISSUES CLOSED: #1152

**What kind of change does this PR introduce?**
Added default sw in create command template

**Did you add tests for your changes?**
No

**If relevant, did you update the documentation?**
No

**Summary**
* add default sw in create command scaffold which was missing before

**Does this PR introduce a breaking change?**
No

**Other information**

### CODE REVIEW

**Comment by anshumanv** [comment]
When do we update the docs regarding the `init` -> `create` command migration? Everything still says `init` for now, should I update those too?

**Comment by rishabh3112** [comment]
@anshumanv main documentation should be updated after public release of v4 IMO as `create` is not supported by v3.
A migration guide in next branch `README` would be the fine solution. I am opening an issue for this, If you like to work on it, send a separate PR for this.

**Comment by anshumanv** [comment]
> A migration guide in next branch README would be the fine solution. I am opening an issue for this, If you like to work on it, send a separate PR for this.

Will send a separate one.

**Comment by anshumanv** [comment]
Fixing conflicts

File: packages/generators/init-generator.ts [hunk 1]
```diff
@@ -28,274 +19,212 @@ import { WebpackOptions } from "./types";
...
-			this.dependencies.push("html-webpack-plugin");
-			const htmlWebpackDependency = "html-webpack-plugin";
-			const htmlwebpackPlugin = generatePluginName(htmlWebpackDependency);
-			(this.configuration.config.topScope as string[]).push(
-				`const ${htmlwebpackPlugin} = require('${htmlWebpackDependency}')`,
-				"\n",
-				tooltip.html()
-			);
-			(this.configuration.config.webpackOptions.plugins as string[]).push(`new ${htmlwebpackPlugin}({
+    public usingDefaults: boolean;
```
Comment (ematipico) [inline packages/generators/init-generator.ts]: Could you please point us where exactly you made some changes? It's difficult to understand

Comment (anshumanv) [reply packages/generators/init-generator.ts]: Sure will do once I fix the conflicts, turns out my editor formatted these files on save taking into account the .editorconfig and .prettierrc which contains massive line length(233), would add better rules to prettier config in a separate PR.

File: packages/generators/init-generator.ts [hunk 2]
```diff
@@ -28,274 +19,212 @@ import { WebpackOptions } from "./types";
...
+            Object.keys(entry).forEach((name: string): void => generateEntryFile(entry[name], `${name} main file!`));
+        }
+
+        // Generate README
+        this.fs.copyTpl(path.resolve(__dirname, './templates/README.md'), this.destinationPath('README.md'), {});
+
+        // Generate HTML template file, copy default service worker
+        if (this.usingDefaults) {
+            this.fs.copyTpl(path.resolve(__dirname, './templates/template.html'), this.destinationPath('index.html'), {});
+            this.fs.copyTpl(path.resolve(__dirname, './templates/sw.js'), this.destinationPath('sw.js'), {});
```
Comment (anshumanv) [inline packages/generators/init-generator.ts]: Copied over the sw file as well on default scaffold, this was already imported in the template html file but the sw file was absent so added it.

File: packages/generators/templates/sw.js [hunk 3]
```diff
@@ -0,0 +1,7 @@
+self.addEventListener('install', event => {
```
Comment (anshumanv) [inline packages/generators/templates/sw.js]: Created a simple sw file here.

**Comment by anshumanv** [comment]
> Please create a seperate PR with lint changes. In this PR, keep changes corresponding to the title.

Yes, reverting along with conflicts.

**Comment by anshumanv** [comment]
Should be good now! PTAL @ematipico @rishabh3112 😄

**Comment by rishabh3112** [comment]
Thanks!

**Pull request merged**